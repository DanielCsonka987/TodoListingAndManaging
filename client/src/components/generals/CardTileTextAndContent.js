import React from 'react'

function CardTileTextAndContent(props){

    const cardStructure =  <p className={'clickable iconTextTogether ' + props.wrapperInlineClasses || ''} 
        onKeyPress={props.funcKeyPressActivity} onClick={props.funcClickActivity}
        tabIndex={props.tabIndexing}>
        <i className='material-icons iconItself'>{props.iconDef}</i>
        <span className={'textNextToIcon tileNameText ' + props.textClasses}>
            {props.tileText}
        </span>
</p>
    if(props.setTestId){
        return (
            <div data-testid={props.setTestId}
                className={'cardAreaPadding cardAreaEdge ' + props.wrapperBlockClasses || ''}>
                { cardStructure }
                { props.children }
            </div>
        )
    }else{
        return(
            <div className={'cardAreaPadding cardAreaEdge ' + props.wrapperBlockClasses || ''}>
                { cardStructure }
                { props.children }
            </div>
        )
    }

}

export default CardTileTextAndContent