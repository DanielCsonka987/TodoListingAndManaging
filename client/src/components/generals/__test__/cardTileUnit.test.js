import React from 'react'
import { waitFor, fireEvent, screen, render } from '@testing-library/react'
import CardTileTextAndContent from '../CardTileTextAndContent'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer';
import { toMatchDiffSnapshot } from 'snapshot-diff'
import { expect as chaiExpect } from 'chai';
import userEvent from '@testing-library/user-event'

describe('Card entity tests', ()=>{
    test('Normal assembling', ()=>{
        const funcForClicking = jest.fn()
        const funcForTyping = jest.fn()
        const cardHeader = 'Test Tile Text'
        const cardIdentifClass = 'stgToTest'
        const cardContent = 'Test text for card content'
        const iconDefinition = 'cancel'

        const inner = <p className={cardIdentifClass}>{cardContent}</p>
        const comp = <CardTileTextAndContent 
            wrapperBlockClasses='cardStyleClass'
            wrapperInlineClasses='cardContnetnClass'
            funcClickActivity={funcForClicking}
            funcKeyPressActivity={funcForTyping}
            iconDef={iconDefinition} tabIndexing='0'
            tileText={cardHeader}
        >
            <span>{inner}</span>
        </CardTileTextAndContent>

        const tree = renderer.create(comp).toJSON()
        expect(tree).toMatchSnapshot()

        const { container, debug, unmount } = render(comp)

        const baseTag = container.firstChild
        expect(baseTag).toHaveClass('cardArea')
        expect(baseTag).toHaveClass('profileCardWidth')
        expect(baseTag).toHaveClass('wrapperColumnAllCenter')
        expect(baseTag).toHaveClass('cardStyleClass')

        //header examination
        const iconTileTextWrapper = baseTag.firstChild
        expect(iconTileTextWrapper).toBeInTheDocument()
        expect(iconTileTextWrapper).toHaveClass('iconTextTogether')
        expect(iconTileTextWrapper).toHaveClass('cardContnetnClass')

        const iTag = screen.queryByText(iconDefinition)
        expect(iTag).toBeInTheDocument()
        expect(iTag).toHaveClass('material-icons')
        expect(iTag).toHaveClass('iconItself')

        const headerText = screen.queryByText(cardHeader)
        expect(headerText).toBeInTheDocument()
        expect(headerText).toHaveClass('textNextToIcon')
        expect(headerText).toHaveClass('tileNameText')

        //content examination
        const innerContent = screen.queryByText(cardContent)
        expect(innerContent).toBeInTheDocument()
        expect(innerContent).toHaveClass(cardIdentifClass)

        unmount()
    })

    test('Selection of card test', ()=>{

        const listOfTileCards = <div>
            <CardTileTextAndContent 
                tabIndexing='0'
                tileText='firstHeader'
            >
                <span>{'First'}</span>
            </CardTileTextAndContent>
            <CardTileTextAndContent 
                tabIndexing='0'
                tileText='secondHeader'
            >
                <span>{'Second'}</span>
            </CardTileTextAndContent>
            <CardTileTextAndContent 
                tabIndexing='-1'
                tileText='thirdHeader'
            >
                 <span>{'Third'}</span>
            </CardTileTextAndContent>
            <CardTileTextAndContent 
                tabIndexing='0'
                tileText='forthHeader'
            >
                <span>{'Forth'}</span>
            </CardTileTextAndContent>
        </div>

        const { container, debug, unmount} = render(listOfTileCards)

        expect(document.body).toHaveFocus()
        userEvent.tab()
        expect(screen.getByText('firstHeader').parentElement).toHaveFocus()
        userEvent.tab()
        expect(screen.getByText('secondHeader').parentElement).toHaveFocus()
        userEvent.tab()
        expect(screen.getByText('forthHeader').parentElement).toHaveFocus()
        userEvent.tab()
        expect(document.body).toHaveFocus()
        userEvent.tab()
        expect(screen.getByText('firstHeader').parentElement).toHaveFocus()

        unmount()
    })
    

})