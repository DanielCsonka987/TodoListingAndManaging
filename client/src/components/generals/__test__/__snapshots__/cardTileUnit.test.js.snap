// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card entity tests Normal assembling 1`] = `
<div
  className="cardAreaPadding cardAreaEdge cardStyleClass"
>
  <p
    className="clickable iconTextTogether cardContnetnClass"
    onClick={[MockFunction]}
    onKeyPress={[MockFunction]}
    tabIndex="0"
  >
    <i
      className="material-icons iconItself"
    >
      cancel
    </i>
    <span
      className="textNextToIcon tileNameText undefined"
    >
      Test Tile Text
    </span>
  </p>
  <span>
    <p
      className="stgToTest"
    >
      Test text for card content
    </p>
  </span>
</div>
`;
